Sources/UI/Chat/Attachments/AttachmentPreview.swift

```diff
@@ -1,353 +1,31 @@
 //
-//  MessageAttachmentPreviewView.swift
+//  AttachmentPreviewProtocol.swift
 //  StreamChat
 //
-//  Created by Alexey Bukhtin on 09/04/2019.
+//  Created by Alexey Bukhtin on 12/04/2019.
 //  Copyright © 2019 Stream.io Inc. All rights reserved.
 //
 
 import UIKit
 import StreamChatClient
 import StreamChatCore
 import RxSwift
-import RxCocoa
-import SnapKit
-import Nuke
-import SwiftyGif
 
-final class AttachmentPreview: UIView, AttachmentPreviewProtocol {
+protocol AttachmentPreview where Self: UIView {
+    typealias Completion = (AttachmentPreview, Error?) -> Void
     
-    private var defaultHeight: CGFloat = .attachmentPreviewMaxHeight
+    /// An index in the message stack view.
+    var index: Int { get set }
+    /// An attachment.
+    var attachment: Attachment? { get }
+    var disposeBag: DisposeBag { get }
```

Comment:
Why do we need disposeBag as a protocol dependency?

Reply:
Yes, we no need it in the protocol.